src/instance/mod.rs

```diff
@@ -358,41 +415,64 @@ impl Instance {
         let position = entity.get_position();
         let skin = entity.get_skin();
         let pv = entity.get_pv();
+
         if self.actors.assign_entity_to_actor(id, entity_id) {
             entity.set_actor(Some(id));
             self.entities.push(entity);
             let notification = Notification::new_entity(entity_id.as_u64(), position, skin, pv);
             self.next_notifications.push(notification);
         } else {
-            // Could be normal operation if the actor has just been unregistered (race
-            // condition)
+            // We are subject to a race condition here. It is possible
+            // that an actor has been removed after the request to
+            // assign to it an entity has been raised but before this
+            // request has been processed by the instance.  In such a
+            // case, the entity should be send back to the Game
+            // instance.
+            //
+            // See https://github.com/greenpix/lycan/...
             warn!("Missing actor {} when sending entity {}", id, entity.get_id());
-            // TODO: Should send back to the Game
         }
+
+        // This should been removed or at least disable by default
+        //
+        // See https://github.com/greenpix/lycan/...
         debug!("{}", self);
     }
 
     pub fn get_id(&self) -> Id<Self> {
         self.id
     }
 
-    fn calculate_tick(&mut self) {
-        trace!("Instance {}: Calculating tick\n{}", self.id, self);
+    fn compute_tick(&mut self) {
+        trace!("Instance {}: Computing tick\n{}", self.id, self);
+
         self.actors.execute_orders(&mut self.entities,
                                    &mut self.next_notifications,
                                    &self.prev_notifications);
 
+        // we process each events, that is something that occurs
+        // between two ticks and need to be handle by the
+        // instance. For example, an entity which executes a step
```

Comment:
Yes. The current idea is to have actors give "high level orders" like "start moving in that direction" or "start attacking", and then the core logic of how this happens (collision checks etc.) is done in `src/entity/update`